cpp/ql/src/experimental/Security/CWE/CWE-409-DecompressionBomb/DecompressionBombs.ql

```diff
@@ -0,0 +1,186 @@
+      fc.getArgument(0) = sink.asExpr()
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof Mz_zip_reader_entry |
+      fc.getArgument(1) = sink.asExpr()
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof Mz_zip_entry |
+      fc.getArgument(1) = sink.asExpr()
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof LzmaStreamBufferDecodeFunction |
+      fc.getArgument(1) = sink.asExpr()
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof LzmaCodeFunction |
+      fc.getArgument(0) = sink.asExpr()
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof GzReadFunction |
+      fc.getArgument(0) = sink.asExpr()
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof GzFreadFunction |
+      sink.asExpr() = fc.getArgument(3)
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof GzGetsFunction |
+      sink.asExpr() = fc.getArgument(0)
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof InflateFunction |
+      fc.getArgument(0) = sink.asExpr()
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof UncompressFunction |
+      fc.getArgument(0) = sink.asExpr()
+    )
+  }
+
+  predicate isAdditionalFlowStep(DataFlow::Node node1, DataFlow::Node node2) {
+    exists(FunctionCall fc | fc.getTarget() instanceof UnzOpenFunction |
+      node1.asExpr() = fc.getArgument(0) and
+      node2.asExpr() = fc
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof Mz_zip_reader_entry |
+      node1.asExpr() = fc.getArgument(0) and
+      node2.asExpr() = fc.getArgument(1)
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof Mz_zip_entry |
+      node1.asExpr() = fc.getArgument(0) and
+      node2.asExpr() = fc.getArgument(1)
+    )
+    or
+    exists(FunctionCall fc |
+      fc.getTarget() instanceof GzopenFunction or fc.getTarget() instanceof GzdopenFunction
+    |
+      node1.asExpr() = fc.getArgument(0) and
+      node2.asExpr() = fc
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof GzReadFunction |
+      node1.asExpr() = fc.getArgument(0) and
+      node2.asExpr() = fc.getArgument(1)
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof GzFreadFunction |
+      node1.asExpr() = fc.getArgument(3) and
+      node2.asExpr() = fc.getArgument(0)
+    )
+    or
+    exists(FunctionCall fc | fc.getTarget() instanceof GzGetsFunction |
+      node1.asExpr() = fc.getArgument(0) and
+      node1.asExpr() = fc.getArgument(1)
+    )
+  }
+}
+
+module DecompressionTaint = TaintTracking::Global<DecompressionTaintConfig>;
+
+import DecompressionTaint::PathGraph
+
+from DecompressionTaint::PathNode source, DecompressionTaint::PathNode sink
+where DecompressionTaint::flowPath(source, sink)
+select sink.getNode(), source, sink, "This Decompression output $@.", source.getNode(),
+  "is not limited"
```

Comment:
Why will the examples from you `example_good.cpp` file not be flagged up by this query?